junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/support/TypeBasedParameterResolver.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.support;
```

Comment:
Sorry, my bad, I should have been more specific. This class should definitely be in jupiter-api and I thought we had already introduced a support package there when we added `@TempDir`. But I was wrong.

@junit-team/junit-lambda How about a new `org.junit.jupiter.api.extension.support` package?

Reply:
sure

Reply:
inside the junit-jupiter-api module ?

Reply:
> Sorry, my bad, I should have been more specific. This class should definitely be in jupiter-api and I thought we had already introduced a support package there when we added `@TempDir`. But I was wrong.
> 
> @junit-team/junit-lambda How about a new `org.junit.jupiter.api.extension.support` package?

SGTM 👍

Reply:
> inside the junit-jupiter-api module ?

Yes, so that implementors won't have to add a compile dependency on junit-jupiter-engine.

Reply:
done